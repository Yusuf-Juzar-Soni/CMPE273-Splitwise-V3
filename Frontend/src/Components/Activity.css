.row {
    width: 100;
    height: 75%;
  }
  .links-acttivity-groups{
      align-items: center;

  }