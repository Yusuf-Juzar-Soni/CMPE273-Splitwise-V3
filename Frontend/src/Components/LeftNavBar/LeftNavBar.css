.flex-column{
background-color:rgb(246, 252, 246) !important;
height: 100vh;
width: 200px;     /* Set the width of the sidebar */
z-index: 1;      /* Stay on top of everything */
     /* Stay at the top */
padding-top: 10px;

}
