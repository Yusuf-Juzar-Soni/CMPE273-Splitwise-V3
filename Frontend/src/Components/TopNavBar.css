.dtoggle{
    margin-right: 109px;
}
