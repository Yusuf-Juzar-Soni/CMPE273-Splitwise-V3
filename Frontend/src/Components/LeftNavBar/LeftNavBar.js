import { React, useEffect, useState } from "react";
import Nav from "react-bootstrap/Nav";
import axios from "axios";
import { useHistory } from "react-router-dom";
import { useSelector } from "react-redux";
const queryString = require("query-string");
import { useLocation } from "react-router-dom";
import "./LeftNavBar.css";
import {
  Button,
  Grid,
  Row,
  Col,
  ListGroup,
  Form,
  Card,
  Modal,
} from "react-bootstrap";
import backendServer from "../../webConfig";

function LeftNavBar() {
  const history = useHistory();
  const emailID = useSelector((state) => state.isLogged.email);

  const [group_names, group_namesChange] = useState([]);
  const [change, setHandleChange] = useState(false);
  const [list, setListDisplay] = useState([]);
  const location = useLocation();
  const parsed = queryString.parse(location.search);
  function retriveGroups(user_email1) {
    const user_email = { email: user_email1 };
    axios
      .post(`${backendServer}/dashboard`, user_email)
      .then((response) => {
        console.log(response.data);
        group_namesChange(response.data);
        console.log(group_names);
      })
      .catch((err) => {
        if (err.response && err.response.data) {
          alert(err.response.data.message);
        }
      });
  }

  function redirectToGroup(redirectGroup, emailId) {
    history.push({
      pathname: "/groupsdisplay",
      search: `?groupname=` + redirectGroup + `&email=` + emailId,
    });
  }

  function redirectToDashboard(emailId) {
    history.push({
      pathname: "/dash",
      search: `?email=${emailId}`,
    });
  }

  function redirectToCreate(emailId) {
    history.push({
      pathname: "/creategroup",
      search: `?email=${emailId}`,
    });
  }

  function redirectToActivity(emailId) {
    history.push({
      pathname: "/activitydisplay",
      search: `?email=${emailId}`,
    });
  }

  function redirectToInvites(emailId) {
    history.push({
      pathname: "/invitedisplay",
      search: `?email=${emailId}`,
    });
  }
  const handleSearch = (e, groups) => {
    setHandleChange(true);
    console.log(groups);
    let final_list = groups.filter((names) => {
      return names.includes(e.target.value);
    });
    setListDisplay(final_list);
  };

  useEffect(() => {
    retriveGroups(parsed.email);
  }, [location]);

  return (
    <div>
      <div>
        <Nav defaultActiveKey="/home" className="flex-column" color="red">
          <Nav.Link onClick={(event) => redirectToDashboard(parsed.email)}>
            Dashboard
          </Nav.Link>
          <Nav.Link
            data-testid="Activity"
            onClick={(event) => redirectToActivity(parsed.email)}
          >
            Activity
          </Nav.Link>
          <Nav.Link>Groups</Nav.Link>
          <input
            type="text"
            onChange={(e) => handleSearch(e, group_names)}
          ></input>
          {change == true && (
            <ListGroup className="list-group-design">
              {list.map((item) => (
                <Button
                  value={item}
                  key={item}
                  variant="link"
                  onClick={(event) =>
                    redirectToGroup(event.currentTarget.value, parsed.email)
                  }
                >
                  {item}
                </Button>
              ))}
            </ListGroup>
          )}

          {change == false && (
            <ListGroup className="list-group-design">
              {group_names.map((item) => (
                <Button
                  value={item}
                  key={item}
                  variant="link"
                  onClick={(event) =>
                    redirectToGroup(event.currentTarget.value, parsed.email)
                  }
                >
                  {item}
                </Button>
              ))}
            </ListGroup>
          )}
          <Nav.Link onClick={(event) => redirectToCreate(parsed.email)}>
            Create Group
          </Nav.Link>
          <Nav.Link onClick={(event) => redirectToInvites(parsed.email)}>
            Invite List
          </Nav.Link>
        </Nav>
      </div>
    </div>
  );
}
export default LeftNavBar;
