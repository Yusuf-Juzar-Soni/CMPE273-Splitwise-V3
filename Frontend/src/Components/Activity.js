import React, { useEffect, useState } from "react";
import LeftNavBar from "./LeftNavBar/LeftNavBar";
import Navbar from "react-bootstrap/Navbar";
import bg_image0 from "./assets/login_logo.png";
import { useSelector } from "react-redux";
import NavDropdown from "react-bootstrap/NavDropdown";
import Dropdown from "react-bootstrap/Dropdown";
import Select from "react-select";
import { useHistory, useLocation } from "react-router-dom";
const queryString = require("query-string");
import Axios from "axios";
import TopNavBar from "./TopNavBar";
import "./Activity.css";
import {
  Nav,
  Row,
  Col,
  Button,
  Card,
  Container,
  ListGroup,
  Modal,
  Form,
} from "react-bootstrap";
import backendServer from "../webConfig";

const Activity = () => {
  const location = useLocation();
  const isLogged = useSelector((state) => state.isLogged.username);
  const [activity, setActivity] = useState([]);
  const parsed = queryString.parse(location.search);
  const email = parsed.email;

  useEffect(() => {
    Axios.get(`${backendServer}/Activity/` + email)
      .then((response) => {
        console.log(response);
        setActivity(response.data);
        console.log(activity);
      })
      .catch((e) => {
        console.log(e);
      });
  }, [location]);

  return (
    <div>
      <div>
        <TopNavBar />
      </div>
      <div className="row">
        <div div className="col-md-2">
          <LeftNavBar />
        </div>
        <div className="col-md-10">
          <h4 data-testid="Activity">Recent Activity</h4>
          <br></br>
          <div className="row ml-1">
            <div className="col-sm-12">
              <ListGroup variant="flush">
                {activity.map((activity) => (
                  <ListGroup.Item
                    variant="info"
                    // key={bill.amount}
                    className="links-acttivity-groups"
                  >
                    <b>{activity.created_by}</b> &nbsp;paid&nbsp;{" "}
                    <b>{activity.bill_amount}</b>
                    &nbsp; in &nbsp; <b>{activity.bill_group}</b>&nbsp;for&nbsp;
                    <b>{activity.bill_desc}</b>&nbsp;on&nbsp;
                    <b>{activity.bill_timestamp}</b>
                    <br></br>
                  </ListGroup.Item>
                ))}
              </ListGroup>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Activity;
