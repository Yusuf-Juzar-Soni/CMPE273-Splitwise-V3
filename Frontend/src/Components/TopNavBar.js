import React from "react";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";
import Dropdown from "react-bootstrap/Dropdown";
import { useHistory } from "react-router-dom";
import { ListGroup, Button } from "react-bootstrap";
import { useSelector } from "react-redux";
const queryString = require("query-string");
import { useLocation } from "react-router-dom";
import bg_image0 from "./assets/login_logo.png";
import "./TopNavBar.css";

function TopNavBar() {
  const history = useHistory();
  const location = useLocation();
  const parsed = queryString.parse(location.search);
  const isLogged = useSelector((state) => state.isLogged.username);

  function redirectToProfile(emailId) {
    history.push({
      pathname: "/profile",
      search: `?email=` + emailId,
    });
  }

  function redirectToLogin() {
    localStorage.clear();
    history.push({
      pathname: "/login",
    });
  }
  return (
    <div>
      <Navbar bg="success" expand="lg">
        <Navbar.Brand href="#home">
          <img src={bg_image0} width="50" height="50"></img>
        </Navbar.Brand>
        <Nav className="mr-auto">
          <Nav.Link href="#home" />
        </Nav>
        <Dropdown>
          <Dropdown.Toggle
            className="dtoggle"
            variant="info"
            id="dropdown-basic"
          >
            {isLogged}
          </Dropdown.Toggle>
          <Dropdown.Menu>
            <Dropdown.Item
              className="ditem"
              onClick={(event) => redirectToProfile(parsed.email)}
            >
              Profile
            </Dropdown.Item>
            <Dropdown.Item onClick={(event) => redirectToLogin()}>
              Logout
            </Dropdown.Item>
          </Dropdown.Menu>
        </Dropdown>
      </Navbar>
    </div>
  );
}

export default TopNavBar;
