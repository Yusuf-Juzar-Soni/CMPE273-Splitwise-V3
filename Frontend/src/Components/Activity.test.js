import React from "react";
import { render, cleanup } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import { createStore } from "redux";
import { MemoryRouter } from "react-router-dom";
// import configureStore from 'redux-mock-store';
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import combReducers from "../reducers";
import TopNavBar from "./TopNavBar";
import LeftNavBar from "./LeftNavBar/LeftNavBar"
import Activity from"./Activity"

const store = createStore(combReducers);

const component = TestRenderer.create(
  <Provider store={store}>
    <MemoryRouter>
      <Activity>
        <TopNavBar />
        <LeftNavBar></LeftNavBar>
        </Activity>
    </MemoryRouter>
  </Provider>
);

afterEach(cleanup);

it("renders", async () => {
  expect(component.toJSON()).toMatchSnapshot();
});

test("Check for header", async () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Activity>
          <TopNavBar />
          
        </Activity>
        
      </MemoryRouter>
    </Provider>
  );
  expect(getByTestId("Activity")).toHaveTextContent("Recent Activity");
});
