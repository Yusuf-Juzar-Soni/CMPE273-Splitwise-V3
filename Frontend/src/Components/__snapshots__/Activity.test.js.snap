// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div>
    <div>
      <nav
        className="navbar navbar-expand-lg navbar-light bg-success"
      >
        <a
          className="navbar-brand"
          href="#home"
        >
          <img
            height="50"
            src="login_logo.png"
            width="50"
          />
        </a>
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rb-event-key="#home"
            href="#home"
            onClick={[Function]}
            onKeyDown={[Function]}
          />
        </div>
        <div
          className="dropdown"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dtoggle dropdown-toggle btn btn-info"
            disabled={false}
            id="dropdown-basic"
            onClick={[Function]}
            type="button"
          >
            
          </button>
        </div>
      </nav>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-2"
      div={true}
    />
    <div
      className="col-md-10"
    >
      <h4
        data-testid="Activity"
      >
        Recent Activity
      </h4>
      <br />
      <div
        className="row ml-1"
      >
        <div
          className="col-sm-12"
        >
          <div
            className="list-group list-group-flush"
            onKeyDown={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
